src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -203,7 +175,7 @@ private unsafe Socket(SafeSocketHandle handle, bool loadPropertiesFromHandle)
                         break;
 
                     case AddressFamily.Unix:
-                        socketAddress = new Internals.SocketAddress(_addressFamily, buffer);
+                        socketAddress = new Internals.SocketAddress(_addressFamily, buffer.Slice(0, bufferLength));
```

Comment:
Why don't we need to slice for IPv4 or IPv6 above?

Reply:
I look at the pal and it picks bytes at particular offset, ignoring the rest as far as I can tell.

Reply:
It'd be more robust to only hand in the relevant portion.

Reply:
I can add it back or add comment. Slice() should be cheap.

Reply:
Let's add it back. Thanks.

Reply:
done.